src/styles/blocks/calendar.scss

```diff
@@ -0,0 +1,55 @@
+.calendar {
+  display: flex;
+  flex-wrap: wrap;
+  gap: 1px;
+  width: $calendar-width;
+  margin: 20px 10px 0;
+
+  @for $days-count from 28 through 31 {
+    &--month-length--#{$days-count} &__box:nth-child(n + #{$days-count + 1}) {
+      display: none;
+    }
+  }
+
+  $days-of-week:
+    "mon" 0 * $size,
+    "tue" 1 * $size + 1px,
```

Comment:
you repeat same calculation each time, it would be better to make this inside loop where you set `margin-left`